packages/protocol/migrations/02_registry.ts

```diff
@@ -1,15 +1,31 @@
-import { deployImplementationAndRepointProxy } from '@celo/protocol/lib/web3-utils'
-import { config } from '@celo/protocol/migrationsConfig'
+import { build_directory, config } from '@celo/protocol/migrationsConfig'
 import { RegistryInstance } from 'types'
+import { setInitialProxyImplementation } from '../lib/web3-utils'
 
-const initializeArgs = async (): Promise<any[]> => {
-  return []
-}
+const Artifactor = require('truffle-artifactor')
+
+const name = 'Registry'
+const Contract = artifacts.require(name)
+const ContractProxy = artifacts.require(name + 'Proxy')
 
-module.exports = deployImplementationAndRepointProxy<RegistryInstance>(
-  web3,
-  artifacts,
-  config.registry.predeployedProxyAddress,
-  'Registry',
-  initializeArgs
-)
+module.exports = (deployer: any, _networkName: string, _accounts: string[]) => {
+  // tslint:disable-next-line: no-console
+  console.log('Deploying Registry')
+  deployer.deploy(ContractProxy)
```

Comment:
I think we don't want to deploy the RegistryProxy, since that one is predeployed in the genesis block, right?

Reply:
i haven't change this. This was the original behaviour..
My understanding is that we deploy it, so we have a `RegitsryProxy.json` in `build/contracts` and we then override the address....
kind of a hack.

i wouldn't do that on mainnet :(

Reply:
I don't think we did a deployment of the RegistryProxy though - take a look at what happens in the old `deployImplementationAndRepointProxy`.

Reply:
we do: 
https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/lib/web3-utils.ts#L318